src/vs/workbench/contrib/chat/browser/chatSetup.ts

```diff
@@ -427,6 +427,10 @@ class ChatSetup {
 			setupStrategy = await this.showDialog();
 		}
 
+		if (setupStrategy === ChatSetupStrategy.DefaultSetup && this.controller.value.hasProviderConfigured()) {
+			setupStrategy = ChatSetupStrategy.SetupWithEnterpriseProvider; // users with a configured provider go through provider setup
```

Comment:
I don't think this is the right change. One of the valid options of this setting is `github` in which case I expect GitHub and not GHE to be used.

I think this should only use GHE if it's set to `github-enterprise`